### PR TITLE
examples: Inject react runtime for ui package

### PR DESCRIPTION
**Pull request by tknickman**
### What has been changed?
1. Update `build` and `dev` command for `kitchen-sink` examples ui package to support components that don't explicitly import `React`.

### Why?
Starting from React 17, React shipped with the new [JSX Transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html), allowing components to include JSX without React being in scope (For example, the `Button` component included below). Because of this, it is now very common to write react components without explicitly importing React at all.

However, in the current `kitchen-sink` example, doing this will break, and throw:
`ReferenceError: React is not defined`.

### How?
In order to support this common use case, we need to instruct `tsup` to inject the `react` runtime into the build. Because `tsup` uses `esbuild`, we can leverage [--inject](https://esbuild.github.io/api/#inject) to automatically replace a global variable with an import from another file. tsup has supported this argument (although poorly documented) since 5.3.0 ([commit](https://github.com/egoist/tsup/commit/f268c649d2ed863aeeb7e77070cab681d2ae8765))

#### Other
* The [esbuild docs cover the exact scenario](https://esbuild.github.io/content-types/#auto-import-for-jsx) this PR is fixing for this example!
* An [issue was raised](https://github.com/egoist/tsup/issues/390) against tsup to support the inject flag to solve this problem as well

Fixing this will hopefully avoid frustration from new users who want to use an example as a starting point, with minimal changes to the build configuration.

### CODE REVIEW

File: examples/kitchen-sink/packages/ui/src/Button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const Button = ({ children, ...other }) => {
```
Comment (jaredpalmer) [inline examples/kitchen-sink/packages/ui/src/Button/index.tsx:1]: Rename to just Button.tsx

File: examples/kitchen-sink/packages/ui/src/CounterButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import * as React from "react";
+
+export const CounterButton = () => {
```
Comment (jaredpalmer) [inline examples/kitchen-sink/packages/ui/src/CounterButton/index.tsx:3]: Move to ./src/CounterButton.tsx

**Review by jaredpalmer — CHANGES_REQUESTED** [review]
Looks, good, just let's get rid of index.tsx's / barrel files aside from the root in `ui`.

File: examples/kitchen-sink/packages/ui/src/Button/index.tsx (hunk 1 above)
Comment (jaredpalmer) [inline examples/kitchen-sink/packages/ui/src/Button/index.tsx:1]: Move to ./src/Button.tsx

**Comment by tknickman** [comment]
@jaredpalmer good call, updated with your suggestions! Thanks!

**Review by jaredpalmer — APPROVED** [review]

**Pull request merged**